### PR TITLE
V2/Address Modal + Transactions Filter

### PR DESCRIPTION
**Pull request by pedrobertao**
Added new Address Modal 
Added Transactions Filter based on the Address Book

### CODE REVIEW

**Comment by pedrobertao** [comment]
wip address modal

File: src/components/ActionModal/elements.js [hunk 1]
```diff
@@ -16,6 +16,12 @@ export const Divider = styled.View`
   height: 1px;
   background-color: ${Colors.background};
 `
+
+export const ActionRow = styled.View`
+  flexDirection: row;
```
Comment (marlonbrgomes) [inline src/components/ActionModal/elements.js]: The pattern is different here from other styles. Please, use kebab case instead.

flex-direction
padding-horizontal
padding-vertical

File: src/components/ButtonIconGradient.js [hunk 2]
```diff
@@ -32,28 +36,33 @@ const ButtonIcon = ({
         colors={[Colors.primaryGradient[0], Colors.primaryGradient[1]]}
         style={[styles.gradientView, {width: size, height: size}]}
       >
-        <View style={[styles.outterView, {width: size * 0.96, height: size * 0.96}]}>
+        <View style={[styles.outterView, {width: size * 0.95, height: size * 0.95, backgroundColor: background || Colors.background}]}>
```
Comment (marlonbrgomes) [inline src/components/ButtonIconGradient.js]: Here we could just use styled component with Colors.background as defaultProps, what are your thoughts about that?

File: src/scenes/Transactions/elements.js [hunk 3]
```diff
@@ -105,3 +106,33 @@ export const EmptyScreenText = styled.Text`
...
+export const FilterText = styled.Text`
+  font-family: Rubik-${props => props.regular ? 'Regular' : 'Medium'};
+  font-size: 14px;
+  line-height: 16px;
+  color: ${props => props.dark ? Colors.secondaryText : Colors.greyBlue};
+  marginHorizontal: ${props => props.marginX ? props.marginX : 0}px;
+`
+export const InnerRow = styled.View`
+  align-items:center;
+  flexDirection: row;
```
Comment (marlonbrgomes) [inline src/scenes/Transactions/elements.js]: Mixing cases here

File: src/scenes/Transactions/elements.js [hunk 4]
```diff
@@ -105,3 +106,33 @@ export const EmptyScreenText = styled.Text`
   letter-spacing: 0;
   color: white;
 `
+
+/* Filter */
+export const FilterWrapper = styled.View`
+  height: 60px;
+  margin-bottom: 20px;
+  margin-horizontal: 20px;
+  backgroundColor: ${Colors.darkThree};
```
Comment (marlonbrgomes) [inline src/scenes/Transactions/elements.js]: mixing typing cases here

File: src/components/ButtonIconGradient.js (hunk 2 above)
Comment (pedrobertao) [reply src/components/ButtonIconGradient.js]: I agree with default props. I actually forgot to write this one.

File: src/components/ButtonIconGradient.js [hunk 5]
```diff
@@ -36,7 +36,7 @@ const ButtonIcon = ({
         colors={[Colors.primaryGradient[0], Colors.primaryGradient[1]]}
         style={[styles.gradientView, {width: size, height: size}]}
       >
-        <View style={[styles.outterView, {width: size * 0.95, height: size * 0.95, backgroundColor: background || Colors.background}]}>
+        <View style={[styles.outterView, {width: size * 0.95, height: size * 0.95, backgroundColor: background}]}>
```
Comment (pedrobertao) [inline src/components/ButtonIconGradient.js]: I don't think we need styled comp here right now. When we start to rewrite to a single pattern we can rework this part of the code @marlonbrgomes

**Pull request merged**